canbus: isotp: fix single frame state machine stuck issue

JiapengLi opened the pull request:
it is too late to set ctx->state to ISOTP_TX_WAIT_FIN after send_sf because send_state_machine could be called just between `send_sf` and `ctx->state = ISOTP_TX_WAIT_FIN;`  in extremely case. like below:
```
 		ret = send_sf(ctx);
		-> irq_handler
		ctx->state = ISOTP_TX_WAIT_FIN;
```
it will cause isotp_send never return.

Fixes: #56754

alexanderwachter reviewed (dismissed):
Good catch!

henrikbrixandersen reviewed (changes requested):
Thank you for providing a fix for this. Can I get you to open a bug issue as well? That will allow us to back port your fix to release branches.

JiapengLi committed:
canbus: isotp: fix single frame state machine stuck issue

it is too late to set ctx->state to ISOTP_TX_WAIT_FIN after send_sf
because send_state_machine could be called just between `send_sf` and
`ctx->state = ISOTP_TX_WAIT_FIN;`  in extremely case. like below:
```c
	ret = send_sf(ctx);
	-> send_state_machine (irq handler)
	ctx->state = ISOTP_TX_WAIT_FIN;
```
it will cause isotp_send never return.

Signed-off-by: Jiapeng Li <mail@jiapeng.me>

JiapengLi commented:
A bug is created. 

The force push to fix commit message format, sorry for the inconvenience.

alexanderwachter commented:
No worries. Sorry for the bug 😅
